Automattic/pocket-casts-ios
# Bookmarks: Add the bookmark indicator to the episode cell

| 🛫 Depends on: https://github.com/Automattic/pocket-casts-ios/pull/1016 | 🎨 Design: [Figma](https://www.figma.com/file/Io7PCz1H1hmOgEE9eljoYW/Bookmarks?type=design&node-id=2033-34038&mode=dev) |
|:---:|:---:|

## Screenshots

| Has Bookmarks | In Up Next | Downloaded | Archived | Downloaded + Up Next |
|:---:|:---:|:---:|:---:|:---:|
|<img width="160" src="https://github.com/Automattic/pocket-casts-ios/assets/793774/c0daf788-aac7-498b-a63e-c336202ceb8a">|<img width="160" src="https://github.com/Automattic/pocket-casts-ios/assets/793774/12c14cb1-f039-4077-a5cb-746892505c71">|<img width="160" src="https://github.com/Automattic/pocket-casts-ios/assets/793774/c9c0e531-1068-4feb-bdc7-93b6ef501dad">|<img width="160"  src="https://github.com/Automattic/pocket-casts-ios/assets/793774/93d7b3fc-f780-4428-b34b-f95cba662992">|<img width="160" src="https://github.com/Automattic/pocket-casts-ios/assets/793774/8b06a6b8-1a57-4513-8593-a47797458ea3">|

## To test

1. Launch the app
2. Enable the Bookmarks Feature Flag
3. Open the Podcast details
4. ✅ Verify any episodes that have bookmarks now show an indicator
5. Play an episode with no bookmarks
6. Create a bookmark for the episode
7. Dismiss the player to view the episode list
8. ✅ Verify it now has a bookmark icon
9. Tap on the episode and swipe to the bookmarks tab
10. Tap and hold, then delete the bookmark
11. Dismiss the episode details
12. ✅ Verify the bookmark indicator is gone
13. Add a bookmark to the episode again
14. Download the episode
15. ✅ Verify the bookmark icon appears after the download icon
16. Add the episode to your up next
17. ✅ Verify the bookmark icon appears after the up next icon
18. Archive the episode
19. ✅ Verify the bookmark icon appears before the archive icon


## Checklist
